json_tokener.c

```diff
@@ -239,7 +239,7 @@ struct json_object* json_tokener_parse_ex(struct json_tokener *tok,
 {
   struct json_object *obj = NULL;
   char c = '\1';
-#ifdef HAVE_USELOCALE
+#if defined(HAVE_USELOCALE) && !defined(__APPLE__)
```

Comment:
If there's some problem with uselocale or duplocale then the configure test should be adjusted to check for that instead.